ambry-cloud/src/main/java/com.github.ambry.cloud/CloudReplica.java

```diff
@@ -72,7 +70,13 @@ public String getReplicaPath() {
 
   @Override
   public List<ReplicaId> getPeerReplicaIds() {
-    return new ArrayList<>(partitionId.getReplicaIds());
+    List<ReplicaId> replicasOfPartition = partitionId.getReplicaIds()
+        .stream()
+        .filter(replica -> replica.getDataNodeId().getPort() != dataNodeId.getPort() || !replica.getDataNodeId()
+            .getHostname()
+            .equals(dataNodeId.getHostname()))
```

Comment:
You can simplify the filter condition to: replica.getDataNodeId().compareTo(dataNodeId) != 0